dialoptions.go

```diff
@@ -446,6 +448,22 @@ func WithDisableServiceConfig() DialOption {
 	})
 }
 
+// WithDefaultServiceConfig returns a DialOption that configures the default service config, which
+// will be used in cases where:
+// 1. WithDisableServiceConfig is called.
+// 2. Resolver does not return service config or if the resolver gets and invalid config.
+func WithDefaultServiceConfig(s string) DialOption {
+	return newFuncDialOption(func(o *dialOptions) {
+		sc, err := parseServiceConfig(s)
+		if err != nil {
+			grpclog.Warningf("the provided service config is invalid, err: %v", err)
+			return
```

Comment:
This is unfortunate. Users won't find out until much later (if at all). Can we provide a ValidateServiceConfig option?

```
sc, err := grpc.ValidateServiceConfig("...")
if err != nil {
    panic(err)
}
opts = append(opts, grpc.WithDefaultServiceConfig(sc))
```

Reply:
done